src/redux/features/votes/votesSlice.ts

```diff
@@ -0,0 +1,46 @@
+import { ActionReducerMapBuilder, PayloadAction, createSlice } from '@reduxjs/toolkit';
+
+import { StatusT, buildReducers } from 'src/redux/features/asyncUtils';
+import { requestProjectVotesDelete, requestProjectVotesUpdate } from 'src/redux/features/votes/votesAsyncThunks';
+import { ProjectVotesPayload, ProjectVotesResponsePayload } from 'src/types/Votes';
+
+// Define a type for the slice state
+type Data = {
+  error?: string;
+  votes?: ProjectVotesPayload;
+};
```

Comment:
Suggest making this a Record of projectId to ProjectVotesPayload. And use the buildReducers approach, with useArgAsKey param set to true.

Reply:
I was wondering about doing that when I set it up. I peaked in a few slices and it didn't seem like we were doing so in any of the other reducers, at least for projectId.

Reply:
I don't mind making the change, it just seems like it may be inconsistent with the rest of the app?

Reply:
Can you point me to the inconsistency?
Here we are fetching votes by projectId, we need to key the result off the projectId.
Otherwise when we switch to another project, we will end up show stale incorrect data that's in the store, probably relevant for another project.

Reply:
I think I understand what you may have been comparing to.
For org level data, we don't scope the results to any key since we either work one org at a time (or globally such as in the accelerator console).
When operating within an org, storing results by requested id is preferred, [example](https://github.com/terraware/terraware-web/blob/main/src/redux/features/draftPlantingSite/draftPlantingSiteSlice.ts#L43)
There is no global Vote results, these are per project id, hence my request.

Reply:
I was trying to highlight that I couldn't find any instances of `<projectId` in the code base. As far as I can tell we aren't keying off project id anywhere else in the app right now and the change seemed inconsistent. I think the reasoning for it makes sense, just trying to understand why we wanted to make the change here and nowhere else.

Reply:
Will make the change though.

Reply:
It seems like the pattern in a lot of the web app is to reset the data when fetching with new params rather than storing the data in redux keyed off of the params.

Reply:
@joshbuchea we don't reset data, except when user switches orgs.
We have a few patterns, all of them have data stored by id, either directly or indirectly.
In some cases this is built into the response from the BE, example: we list all projects and store the entire list of projects in the redux store. When we want a project by id, we look up this list.
We have cases where we don't list all the data, instead we fetch by id (such as in the example earlier).. in these cases, we key off the id in the store.
We don't store data without knowing what it belongs to.